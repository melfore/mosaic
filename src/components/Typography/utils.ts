type VariantDefaultsType = {
  [key: string]: {
    bottomSpacing: boolean;
    truncated: boolean;
  };
};

const VARIANT_DEFAULTS: VariantDefaultsType = {
  body1: {
    bottomSpacing: false,
    truncated: false,
  },
  caption: {
    bottomSpacing: false,
    truncated: false,
  },
  h5: {
    bottomSpacing: true,
    truncated: true,
  },
  h6: {
    bottomSpacing: true,
    truncated: true,
  },
  overline: {
    bottomSpacing: false,
    truncated: false,
  },
  subtitle1: {
    bottomSpacing: false,
    truncated: false,
  },
};

export const getBottomSpacing = (variant: string, bottomSpacing?: boolean): boolean =>
  bottomSpacing === undefined ? VARIANT_DEFAULTS[variant].bottomSpacing : bottomSpacing;

export const getTruncate = (variant: string, truncated?: boolean): boolean =>
  truncated === undefined ? VARIANT_DEFAULTS[variant].truncated : truncated;

export const VARIANT_COMPONENT_MAP = {
  body1: "p",
  caption: "span",
  h5: "h1",
  h6: "h2",
  overline: "h4",
  subtitle1: "h3",
};
