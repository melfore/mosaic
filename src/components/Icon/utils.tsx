import React from "react";
import MUIAddIcon from "@material-ui/icons/Add";
import MUIDeleteIcon from "@material-ui/icons/Delete";
import MUIEditIcon from "@material-ui/icons/Edit";
import MUIPlaylistAddCheckIcon from "@material-ui/icons/PlaylistAddCheck";
import MUISendIcon from "@material-ui/icons/Send";
import { Icons } from "../../types/Icon";

export const iconsCatalog = {
  [Icons.add]: <MUIAddIcon />,
  [Icons.delete]: <MUIDeleteIcon />,
  [Icons.edit]: <MUIEditIcon />,
  [Icons.playlistAddCheck]: <MUIPlaylistAddCheckIcon />,
  [Icons.send]: <MUISendIcon />,
};
