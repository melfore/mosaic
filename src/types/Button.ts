import { MouseEvent } from "react";
import { BaseType } from "./Base";
import { Icons } from "./Icon";

export enum ButtonIconPosition {
  left = "left",
  right = "right",
}

export interface ButtonIconType {
  name: Icons;
  position?: ButtonIconPosition;
}

export enum ButtonVariants {
  contained = "contained",
  outlined = "outlined",
}

export interface ButtonType extends BaseType {
  elevated?: boolean;
  icon?: ButtonIconType;
  label?: string;
  onClick: (event: MouseEvent) => void;
  variant?: ButtonVariants;
}
