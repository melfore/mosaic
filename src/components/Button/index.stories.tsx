import React from "react";
import { action } from "@storybook/addon-actions";
import { boolean, object, select, text } from "@storybook/addon-knobs";
import { ButtonIconPosition, ButtonVariants } from "../../types/Button";
import { Icons } from "../../types/Icon";
import IntlProviderMock, { LocaleMock, MessageMock } from "../../utils/mocks/IntlProviderMock";
import { DOCS_PAGE_STRUCTURE, StoriesWrapper } from "../../utils/stories";
import Button, { ButtonIntl } from ".";

export default {
  title: "Button",
  component: Button,
  parameters: {
    ...DOCS_PAGE_STRUCTURE,
  },
};

export const Canvas = () => (
  <Button
    dataCy={text("data-cy", "button-identifier")}
    elevated={boolean("elevated", false)}
    icon={object("icon", { name: Icons.send, position: ButtonIconPosition.left })}
    label={text("label", "Example")}
    onClick={action("Click on Button")}
    variant={select("variant", ButtonVariants, ButtonVariants.contained)}
  />
);

export const Elevated = () => (
  <StoriesWrapper>
    <Button dataCy="button" elevated label="Contained" onClick={action("Click on Button")} />
  </StoriesWrapper>
);

export const Disabled = () => (
  <StoriesWrapper>
    <Button dataCy="button" label="Disabled" onClick={action("Click on Button")} disabled />
  </StoriesWrapper>
);

export const Variants = () => (
  <StoriesWrapper>
    <Button dataCy="button" label="Contained" onClick={action("Click on Button")} />
    <Button dataCy="button" label="Outlined" onClick={action("Click on Button")} variant={ButtonVariants.outlined} />
  </StoriesWrapper>
);

export const WithIcon = () => (
  <StoriesWrapper>
    <Button dataCy="button" icon={{ name: Icons.send }} label="Left Icon" onClick={action("Click on Button")} />
    <Button
      dataCy="button"
      icon={{ name: Icons.send, position: ButtonIconPosition.right }}
      label="Right Icon"
      onClick={action("Click on Button")}
      variant={ButtonVariants.outlined}
    />
  </StoriesWrapper>
);

export const WithIntl = () => (
  // IntlProviderMock simulates external IntlProvider context
  // The `ButtonIntl` exported version of `Button` uses `labelId` prop to get a localized label.
  // When using `ButtonIntl` the prop `dataCy` can be omitted as it defaults to `labelId` value.
  <StoriesWrapper>
    <IntlProviderMock locale={select("locale", LocaleMock, LocaleMock.en)}>
      <ButtonIntl labelId={MessageMock.button} onClick={action("Click on Button")} />
    </IntlProviderMock>
  </StoriesWrapper>
);
