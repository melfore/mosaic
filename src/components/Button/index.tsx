import React, { FC } from "react";
import MUIButton from "@material-ui/core/Button";
import Icon from "../Icon";
import { BaseIntlType } from "../../types/Base";
import { ButtonIconPosition, ButtonIconType, ButtonType, ButtonVariants } from "../../types/Button";
import withIntl from "../../utils/hocs/withIntl";

const getIcons = (dataCy: string, iconConfig?: ButtonIconType) => {
  const icons = { endIcon: undefined, startIcon: undefined };
  if (!iconConfig) {
    return;
  }

  const { name, position } = iconConfig;
  const icon = <Icon dataCy={`${dataCy}-icon`} name={name} />;
  if (position === ButtonIconPosition.right) {
    return {
      ...icons,
      endIcon: icon,
    };
  }

  return {
    ...icons,
    startIcon: icon,
  };
};

/**
 * Button component made on top of `@material-ui/core/Button`.
 */
const Button: FC<ButtonType> = ({
  dataCy = "button",
  elevated = false,
  icon = undefined,
  label = "",
  onClick,
  variant = ButtonVariants.contained,
  disabled = false,
}) => {
  return (
    <MUIButton
      color="primary"
      data-cy={dataCy}
      disableElevation={!elevated}
      onClick={onClick}
      variant={variant}
      {...getIcons(dataCy, icon)}
      disabled={disabled}
    >
      {label}
    </MUIButton>
  );
};

export const ButtonIntl: FC<ButtonType & BaseIntlType> = withIntl(Button);

export default Button;
