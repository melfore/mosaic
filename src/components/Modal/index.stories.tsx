import React from "react";
import { action } from "@storybook/addon-actions";
import { text, object, boolean, select } from "@storybook/addon-knobs";
import { ButtonVariants } from "../../types/Button";
import ModalMock from "../../utils/mocks/ModalMock";
import { DOCS_PAGE_STRUCTURE, StoriesWrapper } from "../../utils/stories";
import Modal, { ModalIntl } from ".";
import IntlProviderMock, { LocaleMock, MessageMock } from "../../utils/mocks/IntlProviderMock";

export default {
  title: "Modal",
  component: Modal,
  parameters: {
    ...DOCS_PAGE_STRUCTURE,
  },
};

export const Canvas = () => (
  <ModalMock initialOpen={boolean("open", false)} onClose={action("Close action")}>
    <Modal
      cancel={object("cancel", {
        action: action("Optional cancel action"),
        label: "Close",
        variant: ButtonVariants.outlined,
      })}
      closable={boolean("closable", false)}
      confirm={object("confirm", {
        action: action("Optional confirm action"),
        disabled: false,
        label: "Confirm",
      })}
      label={text("label", "Modal Title")}
      onClose={action("Optional close action")}
      open={boolean("open", false)}
    />
  </ModalMock>
);

export const Closable = () => (
  <StoriesWrapper>
    <ModalMock buttonLabel="Closable">
      <Modal closable label="Closable Modal">
        <span>Useful to give another way of dismissal to final user without having actions in modal footer.</span>
      </Modal>
    </ModalMock>
  </StoriesWrapper>
);

export const ModalActions = () => (
  <StoriesWrapper>
    <ModalMock buttonLabel="Default">
      <Modal
        cancel={{ action: () => {}, label: "Cancel", variant: ButtonVariants.outlined }}
        confirm={{ action: () => {}, label: "Confirm" }}
        label="Default Modal"
        onClose={() => {}}
      />
    </ModalMock>
    <ModalMock buttonLabel="Cancel Only">
      <Modal
        cancel={{ action: () => {}, label: "Cancel", variant: ButtonVariants.outlined }}
        label="Cancel Only Modal"
        onClose={() => {}}
      >
        <span>Useful with text only information, for instance explaining why an action can't be done.</span>
      </Modal>
    </ModalMock>
    <ModalMock buttonLabel="Confirm Only">
      <Modal confirm={{ action: () => {}, label: "Confirm" }} label="Confirm Only Modal" onClose={() => {}}>
        <span>Useful when there is no need of having an explicit cancel action.</span>
      </Modal>
    </ModalMock>
    <ModalMock buttonLabel="Confirm Disabled">
      <Modal
        confirm={{ action: () => {}, disabled: true, label: "Confirm" }}
        label="Confirm Disabled Modal"
        onClose={() => {}}
      >
        <span>
          Useful when Confirm action require some input from user, for instance checking out an order. Disable value can
          be controlled from external state to change when user fills required data.
        </span>
      </Modal>
    </ModalMock>
  </StoriesWrapper>
);

export const ScrollableContent = () => (
  <StoriesWrapper>
    <ModalMock buttonLabel="Scrollable Content">
      <Modal closable label="Scrollable Modal">
        <img
          alt="Mosaic"
          src="https://upload.wikimedia.org/wikipedia/commons/thumb/6/60/Roof_hafez_tomb.jpg/440px-Roof_hafez_tomb.jpg"
        />
        <p>
          A mosaic is an artistic picture or design made out of any materials assembled together. Mosaic are used as
          decoration. Architects use mosaic murals for kitchen backsplash, shower wall and entry floor art. Mosaic Craft
          items are used as home decor. Cities often decorate public places such as parks with mosaic murals and
          sculptures.
        </p>
        <p>
          Traditional mosaics are made of small, flat, roughly square pieces of stone or glass of different colors,
          known as tesserae. Some floor mosaics are made of small rounded pieces of stone and called pebble mosaics.
          Mosaic skinning (covering objects with mosaic glass) is done with thin enameled glass and opaque stained
          glass. Modern mosaic art is made from any material in any size ranging from carved stone, bottle caps, and
          found objects.
        </p>
        <p>
          Mosaics have a long history, starting in Mesopotamia in the 3rd millennium BC. Pebble mosaics were made in
          Tiryns in Mycenean Greece; mosaics with patterns and pictures became widespread in classical times, both in
          Ancient Greece and Ancient Rome. Early Christian basilicas from the 4th century onwards were decorated with
          wall and ceiling mosaics. Mosaic art flourished in the Byzantine Empire from the 6th to the 15th centuries;
          that tradition was adopted by the Norman Kingdom of Sicily in the 12th century, by the eastern-influenced
          Republic of Venice, and among the Rus in Ukraine. Mosaic fell out of fashion in the Renaissance, though
          artists like Raphael continued to practise the old technique. Roman and Byzantine influence led Jewish artists
          to decorate 5th and 6th century synagogues in the Middle East with floor mosaics. Mosaic was widely used on
          religious buildings and palaces in early Islamic art, including Islam's first great religious building, the
          Dome of the Rock in Jerusalem, and the Umayyad Mosque in Damascus. Mosaic went out of fashion in the Islamic
          world after the 8th century.
        </p>
        <p>
          Modern mosaics are made by artists and crafters around the world. Many materials other than traditional stone,
          ceramic tesserae, enameled and stained glass may be employed, including shells, beads, charms, chains, gears,
          coins, and pieces of costume jewelry.
        </p>
      </Modal>
    </ModalMock>
  </StoriesWrapper>
);

export const WithIntl = () => (
  <StoriesWrapper>
    <IntlProviderMock locale={select("locale", LocaleMock, LocaleMock.en)}>
      <ModalMock buttonLabel="With Intl">
        <ModalIntl
          cancel={{ labelId: MessageMock.modal_cancel, variant: ButtonVariants.outlined }}
          confirm={{ labelId: MessageMock.modal_confirm }}
          labelId={MessageMock.modal_title}
        >
          <span>Useful to give another way of dismissal to final user without having actions in modal footer.</span>
        </ModalIntl>
      </ModalMock>
    </IntlProviderMock>
  </StoriesWrapper>
);
