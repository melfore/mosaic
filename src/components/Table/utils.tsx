import React, { forwardRef, Fragment } from "react";
import { Action as MTActionType, Column as MTColumnType, Options as MTOptionsType } from "material-table";
import { Icons, IconSize } from "../../types/Icon";
import { TableActionScope, TableActionType, TableColumnType } from "../../types/Table";
import Button from "../Button";
import Icon from "../Icon";
import IconButton from "../IconButton";

// For default props refer to Options section here https://material-table.com/#/docs/all-props
export const DEFAULT_TABLE_OPTIONS: MTOptionsType = {
  debounceInterval: 500,
  draggable: false,
  emptyRowsWhenPaging: false,
};

export const actionAdapter = (action: TableActionType): MTActionType<object> => {
  const {
    callback,
    label,
    disabled = false,
    hidden = false,
    icon = undefined,
    scope = TableActionScope.default,
  } = action;
  return {
    disabled,
    hidden,
    icon: `${label}$${icon}`,
    isFreeAction: scope === TableActionScope.default,
    onClick: callback,
  };
};

export const actionComponentAdapter = (props: any) => {
  const { data, action } = props;
  const { disabled, hidden, icon, isFreeAction, onClick }: MTActionType<object> = action;
  if (hidden) {
    return null;
  }

  const labelAndIconName = icon as string;
  const label = labelAndIconName.split("$")[0];
  const iconName = labelAndIconName.split("$")[1] as Icons;
  return (
    <Fragment>
      {isFreeAction && (
        <Fragment>
          <span>&nbsp;&nbsp;&nbsp;&nbsp;</span>
          <Button
            disabled={disabled}
            icon={!iconName ? undefined : { name: iconName }}
            label={label}
            onClick={(event: any) => {
              onClick(event, data);
              event.preventDefault();
              event.stopPropagation();
            }}
          />
        </Fragment>
      )}
      {!isFreeAction && (
        <Fragment>
          <IconButton
            disabled={disabled}
            icon={iconName}
            onClick={(event: any) => {
              onClick(event, data);
              event.preventDefault();
              event.stopPropagation();
            }}
            size={IconSize.small}
          />
          <span>&nbsp;&nbsp;</span>
        </Fragment>
      )}
    </Fragment>
  );
};

export const columnAdapter = (column: TableColumnType): MTColumnType<object> => {
  const { label, path, render } = column;
  return {
    field: path,
    render,
    title: label,
  };
};

export const iconAdapter = (name: Icons, size = IconSize.default) =>
  forwardRef((props: any, ref: any) => {
    return <Icon forwarded={{ ...props, ref }} name={name} size={size} />;
  });
