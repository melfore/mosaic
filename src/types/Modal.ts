import { MouseEvent } from "react";
import { BaseType } from "./Base";
import { ButtonVariants } from "./Button";

export interface ModalActionType {
  action?: (event: MouseEvent<Element, MouseEvent>) => void;
  label?: string;
  labelId?: string;
}

export interface CancelModalActionType extends ModalActionType {
  variant: ButtonVariants.outlined;
}

export interface ConfirmModalActionType extends ModalActionType {
  disabled?: boolean;
}

export interface ModalType extends BaseType {
  cancel?: CancelModalActionType;
  closable?: boolean;
  confirm?: ConfirmModalActionType;
  label: string;
  onClose?: (event: any, reason?: string) => void;
  open?: boolean;
}
