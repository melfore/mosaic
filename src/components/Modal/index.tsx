import React, { FC } from "react";
import MUIDialog from "@material-ui/core/Dialog";
import MUIDialogActions from "@material-ui/core/DialogActions";
import MUIDialogContent from "@material-ui/core/DialogContent";
import MUIDialogTitle from "@material-ui/core/DialogTitle";
import MUIIconButton from "@material-ui/core/IconButton";
import MUITypography from "@material-ui/core/Typography";
import { styled } from "@material-ui/core/styles";
import Button, { ButtonIntl } from "../Button";
import Icon from "../Icon";
import { Icons } from "../../types/Icon";
import { ModalActionType, ModalType } from "../../types/Modal";

const StyledMUIDialogTitle = styled(MUIDialogTitle)({
  alignItems: "center",
  display: "flex",
  justifyContent: "space-between",
});

const getActionButton = <T extends ModalActionType>(buttonConfig: T, onClose?: Function): any => {
  const { action, label, labelId, ...props } = buttonConfig;
  const onClickHandler = (event: any) => {
    action && action(event);
    onClose && onClose(event);
  };

  if (!labelId) {
    return <Button {...props} label={label} onClick={onClickHandler} />;
  }

  return <ButtonIntl {...props} labelId={labelId} onClick={onClickHandler} />;
};

/**
 * Modal component made on top of `@material-ui/core/Dialog`
 */
const Modal: FC<ModalType> = ({
  cancel,
  children,
  closable = false,
  confirm,
  label,
  onClose = undefined,
  open = false,
}) => {
  const hasActions = cancel || confirm;
  return (
    <MUIDialog aria-labelledby="modal-title" fullWidth maxWidth="sm" onClose={onClose} open={open}>
      <StyledMUIDialogTitle id="modal-title" disableTypography>
        {/* TODO: see #68 */}
        <MUITypography variant="h6">{label}</MUITypography>
        {closable && (
          // TODO: see #69
          <MUIIconButton size="small" onClick={onClose}>
            <Icon name={Icons.close} />
          </MUIIconButton>
        )}
      </StyledMUIDialogTitle>
      <MUIDialogContent dividers>{children}</MUIDialogContent>
      {hasActions && (
        <MUIDialogActions>
          {cancel && getActionButton(cancel, onClose)}
          {confirm && getActionButton(confirm, onClose)}
        </MUIDialogActions>
      )}
    </MUIDialog>
  );
};

export default Modal;
