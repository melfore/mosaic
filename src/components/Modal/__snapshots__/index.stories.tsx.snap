// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feedback/Modal ActionCancel smoke-test 1`] = `
<div>
  <button class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-mui-classname"
          tabindex="0"
          type="button"
          data-cy="button"
  >
    <span class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-mui-classname">
      <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
           focusable="false"
           aria-hidden="true"
           viewbox="0 0 24 24"
           data-testid="OpenInNewIcon"
           data-cy="button-icon"
      >
        <path d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z">
        </path>
      </svg>
    </span>
    Open
  </button>
</div>
`;

exports[`Feedback/Modal ActionConfirm smoke-test 1`] = `
<div>
  <button class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-mui-classname"
          tabindex="0"
          type="button"
          data-cy="button"
  >
    <span class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-mui-classname">
      <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
           focusable="false"
           aria-hidden="true"
           viewbox="0 0 24 24"
           data-testid="OpenInNewIcon"
           data-cy="button-icon"
      >
        <path d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z">
        </path>
      </svg>
    </span>
    Open
  </button>
</div>
`;

exports[`Feedback/Modal Actions smoke-test 1`] = `
<div>
  <button class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-mui-classname"
          tabindex="0"
          type="button"
          data-cy="button"
  >
    <span class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-mui-classname">
      <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
           focusable="false"
           aria-hidden="true"
           viewbox="0 0 24 24"
           data-testid="OpenInNewIcon"
           data-cy="button-icon"
      >
        <path d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z">
        </path>
      </svg>
    </span>
    Open
  </button>
</div>
`;

exports[`Feedback/Modal Closeable smoke-test 1`] = `
<div>
  <button class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-mui-classname"
          tabindex="0"
          type="button"
          data-cy="button"
  >
    <span class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-mui-classname">
      <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
           focusable="false"
           aria-hidden="true"
           viewbox="0 0 24 24"
           data-testid="OpenInNewIcon"
           data-cy="button-icon"
      >
        <path d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z">
        </path>
      </svg>
    </span>
    Open
  </button>
</div>
`;

exports[`Feedback/Modal Localized smoke-test 1`] = `
<div>
  <button class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-mui-classname"
          tabindex="0"
          type="button"
          data-cy="button"
  >
    <span class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-mui-classname">
      <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
           focusable="false"
           aria-hidden="true"
           viewbox="0 0 24 24"
           data-testid="OpenInNewIcon"
           data-cy="button-icon"
      >
        <path d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z">
        </path>
      </svg>
    </span>
    Open
  </button>
</div>
`;

exports[`Feedback/Modal Primary smoke-test 1`] = `
<div>
  <button class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-mui-classname"
          tabindex="0"
          type="button"
          data-cy="button"
  >
    <span class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-mui-classname">
      <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
           focusable="false"
           aria-hidden="true"
           viewbox="0 0 24 24"
           data-testid="OpenInNewIcon"
           data-cy="button-icon"
      >
        <path d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z">
        </path>
      </svg>
    </span>
    Open
  </button>
</div>
`;

exports[`Feedback/Modal SizeFullScreen smoke-test 1`] = `
<div>
  <button class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-mui-classname"
          tabindex="0"
          type="button"
          data-cy="button"
  >
    <span class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-mui-classname">
      <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
           focusable="false"
           aria-hidden="true"
           viewbox="0 0 24 24"
           data-testid="OpenInNewIcon"
           data-cy="button-icon"
      >
        <path d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z">
        </path>
      </svg>
    </span>
    Open
  </button>
</div>
`;

exports[`Feedback/Modal SizeLarge smoke-test 1`] = `
<div>
  <button class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-mui-classname"
          tabindex="0"
          type="button"
          data-cy="button"
  >
    <span class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-mui-classname">
      <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
           focusable="false"
           aria-hidden="true"
           viewbox="0 0 24 24"
           data-testid="OpenInNewIcon"
           data-cy="button-icon"
      >
        <path d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z">
        </path>
      </svg>
    </span>
    Open
  </button>
</div>
`;

exports[`Feedback/Modal SizeSmall smoke-test 1`] = `
<div>
  <button class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-disableElevation css-mui-classname"
          tabindex="0"
          type="button"
          data-cy="button"
  >
    <span class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-mui-classname">
      <svg class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mui-classname"
           focusable="false"
           aria-hidden="true"
           viewbox="0 0 24 24"
           data-testid="OpenInNewIcon"
           data-cy="button-icon"
      >
        <path d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3z">
        </path>
      </svg>
    </span>
    Open
  </button>
</div>
`;
